hist/histpainter/src/THistPainter.cxx

```diff
@@ -3739,42 +3739,43 @@ char *THistPainter::GetObjectInfo(Int_t px, Int_t py) const
    if (fH->GetDimension() == 1) {
       if (fH->InheritsFrom(TProfile::Class())) {
          TProfile *tp = (TProfile*)fH;
-         snprintf(info,200,"(x=%g, y=%g, binx=%d, binc=%g, bine=%g, binn=%d)",
+         info = TString::Format("(x=%g, y=%g, binx=%d, binc=%g, bine=%g, binn=%d)",
                   x, y, binx, fH->GetBinContent(binx), fH->GetBinError(binx),
                   (Int_t) tp->GetBinEntries(binx));
       }
       else {
          Double_t integ = 0;
          for (Int_t bin=binmin;bin<=binx;bin++) {integ += fH->GetBinContent(bin);}
-         snprintf(info,200,"(x=%g, y=%g, binx=%d, binc=%g, Sum=%g)",
+         info = TString::Format("(x=%g, y=%g, binx=%d, binc=%g, Sum=%g)",
                   x,y,binx,fH->GetBinContent(binx),integ);
       }
    } else if (fH->GetDimension() == 2) {
       if (fH->InheritsFrom(TH2Poly::Class())) {
          TH2Poly *th2 = (TH2Poly*)fH;
          biny = th2->FindBin(x,y);
-         snprintf(info,200,"%s (x=%g, y=%g, bin=%d, binc=%g)",
+         info = TString::Format("%s (x=%g, y=%g, bin=%d, binc=%g)",
                   th2->GetBinTitle(biny),x,y,biny,th2->GetBinContent(biny));
       }
       else if (fH->InheritsFrom(TProfile2D::Class())) {
          TProfile2D *tp = (TProfile2D*)fH;
          biny = fYaxis->FindFixBin(y);
          Int_t bin = fH->GetBin(binx,biny);
-         snprintf(info,200,"(x=%g, y=%g, binx=%d, biny=%d, binc=%g, bine=%g, binn=%d)",
+         info = TString::Format("(x=%g, y=%g, binx=%d, biny=%d, binc=%g, bine=%g, binn=%d)",
                   x, y, binx, biny, fH->GetBinContent(bin),
                   fH->GetBinError(bin), (Int_t) tp->GetBinEntries(bin));
       } else {
          biny = fYaxis->FindFixBin(y);
-         snprintf(info,200,"(x=%g, y=%g, binx=%d, biny=%d, binc=%g bine=%g)",
+         info = TString::Format("(x=%g, y=%g, binx=%d, biny=%d, binc=%g bine=%g)",
                   x,y,binx,biny,fH->GetBinContent(binx,biny),
                   fH->GetBinError(binx,biny));
       }
    } else {
       // 3d case: retrieving the x,y,z bin is not yet implemented
       // print just the x,y info
-      snprintf(info,200,"(x=%g, y=%g)",x,y);
+      info = TString::Format("(x=%g, y=%g)",x,y);
    }
-   return info;
+
+   return Form("%s",info.Data());
```

Comment:
It is still not safe to return pointer on global variable, which will be changed with next `Form()` invocation.
See my previous comment - just use `return fObjectIno.Data();` here.